src/mad_spark_alt/core/json_utils.py

```
@@ -182,6 +187,219 @@
+    # Validate recommended_parents
+    if not isinstance(data["recommended_parents"], list):
+        return False
+    for parent_idx in data["recommended_parents"]:
+        if not isinstance(parent_idx, int) or parent_idx < 0:
+            return False
+
+    # Validate diversity_consideration
+    if not _is_valid_non_empty_string(data["diversity_consideration"]):
+        return False
+
+    return True
```

Ruff (0.12.2)

Return the negated condition directly

Inline condition

(SIM103)